Mathlib/MeasureTheory/Function/AEEqOfIntegral.lean

```diff
@@ -629,4 +629,25 @@ theorem AEMeasurable.ae_eq_of_forall_set_lintegral_eq {f g : α → ℝ≥0∞}
 
 end Lintegral
 
+section WithDensity
+
+variable {m : MeasurableSpace α} {μ : Measure α}
+
+theorem withDensity_eq_iff_of_sigmaFinite [SigmaFinite μ] {f g : α → ℝ≥0∞} (hf : Measurable f)
+    (hg : Measurable g) : μ.withDensity f = μ.withDensity g ↔ f =ᵐ[μ] g :=
```

Comment:
I generalized `ae_le_of_forall_set_lintegral_le_of_sigmaFinite` to a.e.-measurable functions in #8032 . Once that PR is merged, you will be able to generalize this lemma as well.